src/CommunityToolkit.Maui.Markup.UnitTests/TypedBindingExtensionsTests.cs

```diff
@@ -0,0 +1,410 @@
+		TaskCompletionSource<string?> sliderPropertyChangedEventArgsTCS = new();
+
+		var slider = new Slider
+		{
+			BindingContext = viewModel
+		}.Bind(Slider.ValueProperty, static (ViewModel viewModel) => viewModel.Percentage, static (ViewModel viewModel, double temperature) => viewModel.Percentage = temperature);
+
+		slider.PropertyChanged += HandleSliderPropertyChanged;
+		viewModel.PropertyChanged += HandleViewModelPropertyChanged;
+
+		BindingHelpers.AssertTypedBindingExists<Slider, ViewModel, string>(slider, Slider.ValueProperty, BindingMode.Default, viewModel);
+		Assert.AreEqual(ViewModel.DefaultPercentage, slider.Value);
+
+		slider.Value = 1;
+		Assert.AreEqual(1, slider.Value);
+		Assert.AreEqual(1, viewModel.Percentage);
+
+		viewModel.Percentage = 0.6;
+		var viewModelPropertyName = await viewModelPropertyChangedEventArgsTCS.Task;
+		var sliderPropertyName = await sliderPropertyChangedEventArgsTCS.Task;
+
+		Assert.True(didViewModelPropertyChangeFire);
+		Assert.AreEqual(nameof(ViewModel.Percentage), viewModelPropertyName);
+		Assert.AreEqual(2, viewModelPropertyChangedEventCount);
+
+		Assert.True(didSliderPropertyChangeFire);
+		Assert.AreEqual(nameof(Slider.Value), sliderPropertyName);
+		Assert.AreEqual(2, sliderPropertyChangedEventCount);
+
+		Assert.AreEqual(0.6, slider.Value);
+		Assert.AreEqual(0.6, viewModel.Percentage);
+
+		void HandleViewModelPropertyChanged(object? sender, PropertyChangedEventArgs e)
+		{
+			didViewModelPropertyChangeFire = true;
+			viewModelPropertyChangedEventCount++;
+			viewModelPropertyChangedEventArgsTCS.TrySetResult(e.PropertyName);
+		}
+
+		void HandleSliderPropertyChanged(object? sender, PropertyChangedEventArgs e)
+		{
+			didSliderPropertyChangeFire = true;
+			sliderPropertyChangedEventCount++;
+			sliderPropertyChangedEventArgsTCS.TrySetResult(e.PropertyName);
+		}
+	}
+
+	[TestCase(false, false)]
+	[TestCase(false, true)]
+	[TestCase(true, false)]
+	[TestCase(true, true)]
+	public void ValueSetOnOneWayWithNestedPathBinding(bool setContextFirst, bool isDefault)
+	{
+		var entry = new Entry();
+		var bindingMode = isDefault ? BindingMode.Default : BindingMode.OneWay;
+
+		var viewmodel = new NestedViewModel
+		{
+			Model = new NestedViewModel
+			{
+				Model = new NestedViewModel()
+			}
+		};
+
+		if (setContextFirst)
+		{
+			entry.BindingContext = viewmodel;
+			entry.Bind(Entry.TextColorProperty,
+							static (NestedViewModel vm) => vm.Model?.Model?.TextColor,
+							new (Func<NestedViewModel, object?>, string)[]
+							{
+								(vm => vm, nameof(NestedViewModel.Model)),
+								(vm => vm.Model, nameof(NestedViewModel.Model)),
+								(vm => vm.Model.Model, nameof(NestedViewModel.Model.TextColor))
+							},
+							static (NestedViewModel vm, Color? color) =>
+							{
+								if (vm.Model?.Model?.TextColor is not null && color is not null)
+								{
+									vm.Model.Model.TextColor = color;
+								}
+							},
+							bindingMode);
```

Comment:
@bijington Here's an example of how to use TypedBindings for nested bindings.

E.g. If there is a Property in the ViewModel that has a Property inside it (like using a Model as a Property in the ViewModel)